File: TASK07/token/token.go
package token

import "fmt"

type Position struct {
	Line   int
	Column int
}

type Token struct {
	Type
	Position
}

type Type int

const (
	Eof Type = iota

	Plus  // +
	Minus // -

	LeftArrow  // <
	RightArrow // >

	Comma  // ,
	Period // .

	LeftBracket  // [
	RightBracket // ]
)

var tokens = [...]string{
	Eof:          "EOF",
	Plus:         "+",
	Minus:        "-",
	LeftArrow:    "<",
	RightArrow:   ">",
	Comma:        ",",
	Period:       ".",
	LeftBracket:  "[",
	RightBracket: "]",
}

func (t Type) String() string {
	return tokens[t]
}

func (pos Position) String() string {
	return fmt.Sprintf("%d:%d", pos.Line, pos.Column)
}

func (pos *Position) NextLine() {
	pos.Line++
	pos.Column = 1
}

File: TASK08/token/token.cpp
#pragma once
#include <iostream>
#include <unordered_map>
#include <string>
using namespace std;

class Position {
public:
    int line;
    int column;

    Position(int line, int col) {
        this->line = line;
        this->column = col;
    }

    string String() {
        return line + ":" + column;
    }

    void NextLine() {
        line++;
        column = 1;
    }
};

class Token {
public:
    int type;
    Position pos;

    Token(int type, int line, int col): pos(line, col) {
        this->type = type;
    }

    string String() {
        string tokens[] = {"EOF", "+", "-", "<", ">", ",", ".", "[", "]"};
        return tokens[type];
    }

    string ParsePos() {
        return pos.column + "-" + pos.line;
    }
};

const int Eof = 0;
const int Plus = 1;
const int Minus = 2;
const int LeftArrow = 3;
const int RightArrow = 4;
const int Comma = 5;
const int Period = 6;
const int LeftBracket = 7;
const int RightBracket = 8;

